### DISCUSSION TITLE
How do we change highlights now?

### DISCUSSION THREAD

**Discussion by olimorris** [utilyre/barbecue.nvim#41] [Q&A]
Just pulled in the latest version and my winbar has turned a shade of `#7a7a7a`. I'm guessing as per the [theme](https://github.com/utilyre/barbecue.nvim/blob/main/lua/barbecue/theme/default.lua).

Previously, I had some lovely highlights that I applied via my colorscheme:

```lua
BarbecueModified = { fg = "${red}" },
BarbecueEllipsis = { fg = "${breadcrumbs}", style = "italic" },
BarbecueSeparator = { fg = "${breadcrumbs}", style = "italic" },
BarbecueDirname = { fg = "${breadcrumbs}", style = "italic" },
BarbecueBasename = { fg = "${breadcrumbs}", style = "italic" },
BarbecueContext = { fg = "${breadcrumbs}", style = "italic" },
```

which worked perfectly. Now it seems as though I will have to create a theme to get back to how I had them before?

Any reason why you can't just expose highlight groups like the majority of other plugins do that we can override?

**Comment by utilyre** [comment]
Refer to these: #39 #30

> Any reason why you can't just expose highlight groups like the majority of other plugins do that we can override?

The highlighting is now done the way lualine does it. The reason being winbar and statusline (in case of lualine) are not just regular windows that you can specify a background for and be done. These "bars" are text based which means if you want a background then you'll have to override every possible highlighting's bg. This might sound like an easy task, just override the several highlights on the (previous) readme and be done. But no, it gets harder when it comes to devicons.

So, as people requested (#30), I did some refactoring and made barbecue create highlight groups dynamically based on `barbecue.theme.[colorscheme]` module. Or the theme table which can be passed to config now (it replaces the mentioned module).

**Reply by olimorris** [reply]
It might be that the readme just needs updating for how to use the new themes module because I can?t ascertain how I move from the previous HL groups to the theme's module.

**Reply by utilyre** [reply]
> It might be that the readme just needs updating for how to use the new themes module because I can?t ascertain how I move from the previous HL groups to the theme's module.

I originally mentioned it on the readme, but I thought this is more on the plugin dev side and eventually moved it to CODE_OF_CONDUCT.md.

**Reply by olimorris** [reply]
So a user who uses `Nightfox` as their colourscheme, has to accept that Barbecue will look awful out of the box until someone makes a PR for it?

Surely, you could pass the colorscheme in as a parameter to the configuration, like in Lazy?

Lualine is not a good comparison imo. It's a widely used plugin that can expect colorscheme authors to conform to its API.

**Reply by utilyre** [reply]
I see. I'll look into making a mechanism for automatically generating a theme when the colorscheme doesn't support this plugin.

**Reply by olimorris** [reply]
For some context, in my [colorscheme](https://github.com/olimorris/onedarkpro.nvim), I support plugins like `nvim_navic` like [so](https://github.com/olimorris/onedarkpro.nvim/blob/main/lua/onedarkpro/highlights/plugins/nvim_navic.lua).

Basically put, it's a helluva lot easier to just expose the highlight groups like you previously did.
